feat: rename types in config

SanderVanRiessen opened the pull request:
This merge request gives you the ability to rename your types, for example if you have conflicts in type names you can use the `as NEWTYPENAME` to work around the type name conflict.

ardeois commented:
@SanderVanRiessen Could you explain why you would need such an option?
[Typescript plugin](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript) doesn't have an option like this and we try to stay close to their configs.

ardeois commented:
@SanderVanRiessen  How is the typescript generator doing in such case? It's generating a `Partial` type as well no?

SanderVanRiessen commented:
Yes the typescript generator is creating a `Partial` type based on the schema. But it is not using the Partial type from Typescript in the generated file.

ardeois commented:
Ok I understand now. It makes sense to have such an option but I've got the following comments:
- I would prefer to use option `typeNamesMapping` instead of `newTypeNames`, please rename it
- Update README to document the new option
- Add unit tests

Thanks !

SanderVanRiessen commented:
Thanks for your comments. 
I have updated the docs with the new option and added new tests, also renamed the option. Let me know if anything else needs to be changed.

ardeois reviewed (approved):
Thanks for the changes 👍
